### ISSUE TITLE
Disconnect after JITP not handled

### ISSUE THREAD

**Issue by coderbyheart** [bifravst/firmware#81]
If the device is connecting for the first time, JITP will cause the connection to be terminated (see https://github.com/nrfconnect/sdk-nrf/pull/3017).

However this is also not handled in our application:

```
*** Booting Zephyr OS build v2.3.0-rc1-ncs1-3614-ga4ead9805140  ***
Flash regions           Domain          Permissions
00 02 0x00000 0x18000   Secure          rwxl
03 31 0x18000 0x100000  Non-Secure      rwxl

Non-secure callable region 0 placed in flash region 2 with size 32.

SRAM region             Domain          Permissions
00 07 0x00000 0x10000   Secure          rwxl
08 31 0x10000 0x40000   Non-Secure      rwxl

Peripheral              Domain          Status
00 NRF_P0               Non-Secure      OK
01 NRF_CLOCK            Non-Secure      OK
02 NRF_RTC0             Non-Secure      OK
03 NRF_RTC1             Non-Secure      OK
04 NRF_NVMC             Non-Secure      OK
05 NRF_UARTE1           Non-Secure      OK
06 NRF_UARTE2           Secure          SKIP
07 NRF_TWIM2            Non-Secure      OK
08 NRF_SPIM3            Non-Secure      OK
09 NRF_TIMER0           Non-Secure      OK
10 NRF_TIMER1           Non-Secure      OK
11 NRF_TIMER2           Non-Secure      OK
12 NRF_SAADC            Non-Secure      OK
13 NRF_PWM0             Non-Secure      OK
14 NRF_PWM1             Non-Secure      OK
15 NRF_PWM2             Non-Secure      OK
16 NRF_PWM3             Non-Secure      OK
17 NRF_WDT              Non-Secure      OK
18 NRF_IPC              Non-Secure      OK
19 NRF_VMC              Non-Secure      OK
20 NRF_FPU              Non-Secure      OK
21 NRF_EGU1             Non-Secure      OK
22 NRF_EGU2             Non-Secure      OK
23 NRF_DPPIC            Non-Secure      OK
24 NRF_GPIOTE1          Non-Secure      OK
25 NRF_REGULATORS       Non-Secure      OK

SPM: NS image at 0x18200
SPM: NS MSP at 0x200315e8
SPM: NS reset vector at 0x21e41
SPM: prepare to jump to Non-Secure image.
*** Booting Zephyr OS build v2.3.0-rc1-ncs1-3614-ga4ead9805140  ***
[00:00:00.200,042] <inf> watchdog: Watchdog timeout installed. Timeout: 60s
[00:00:00.207,580] <inf> watchdog: Watchdog started
[00:00:00.212,982] <inf> watchdog: Watchdog feed enabled. Timeout: 30000
[00:00:00.228,118] <inf> cat_tracker: ********************************************
[00:00:00.236,267] <inf> cat_tracker:  The cat tracker has started
[00:00:00.242,980] <inf> cat_tracker:  Version:     0.0.0-development-thingy91_nrf9160ns-ltem-debug
[00:00:00.252,593] <inf> cat_tracker:  Client ID:   352656100248049
[00:00:00.259,399] <inf> cat_tracker:  Endpoint:    a34x44yyrk96tg-ats.iot.eu-central-1.amazonaws.com
[00:00:00.269,195] <inf> cat_tracker: ********************************************
[00:00:00.277,343] <dbg> nrf9160_gps.open_socket: GPS socket created, fd: 1232491587
[00:00:00.285,705] <inf> cat_tracker: Configuring the modem...
[00:00:01.668,609] <dbg> cat_tracker.modem_rsrp_handler: Incoming RSRP status message, RSRP value is 38
[00:00:01.678,833] <dbg> cat_tracker.modem_rsrp_handler: Incoming RSRP status message, RSRP value is 39
[00:00:01.689,361] <dbg> cat_tracker.lte_evt_handler: LTE cell changed: Cell ID: 35493142, Tracking area: 2305
[00:00:01.700,134] <dbg> cat_tracker.lte_evt_handler: PSM parameter update: TAU: -1, Active time: -1
[00:00:01.755,981] <dbg> cat_tracker.lte_evt_handler: RRC mode: Connected
[00:00:02.861,389] <inf> cat_tracker: Network registration status: Connected to roaming network
[00:00:02.870,727] <dbg> date_time.new_date_time_get: Updating date time UTC...
[00:00:02.878,601] <dbg> date_time.current_time_check: Date time never set
[00:00:02.886,016] <dbg> date_time.new_date_time_get: Current time not valid
[00:00:02.893,615] <dbg> date_time.new_date_time_get: Fallback on cellular network time
[00:00:02.902,374] <dbg> cat_tracker.lte_evt_handler: PSM parameter update: TAU: 108000, Active time: 0
[00:00:02.912,536] <dbg> date_time.time_modem_get: Response from modem: +CCLK: "20/10/07,13:40:36+08
[00:00:02.922,302] <dbg> date_time.new_date_time_get: Time from cellular network obtained
[00:00:02.931,060] <dbg> cat_tracker.date_time_event_handler: DATE_TIME_OBTAINED_MODEM
[00:00:02.939,575] <inf> cat_tracker: Device in ACTIVE mode
[00:00:02.945,831] <inf> cat_tracker: CLOUD_EVT_CONNECTING
[00:00:02.946,411] <dbg> nrf9160_gps.enable_gps: GPS mode is enabled
[00:00:02.965,606] <dbg> nrf9160_gps.gps_priority_set: GPS priority disabled
[00:00:02.973,602] <dbg> nrf9160_gps.start: GPS operational
[00:00:02.979,766] <inf> cat_tracker: GPS_EVT_SEARCH_STARTED
[00:00:02.986,114] <dbg> nrf9160_gps.gps_thread: A-GPS data update needed
[00:00:02.993,469] <inf> cat_tracker: GPS_EVT_AGPS_DATA_NEEDED
[00:00:03.965,301] <dbg> nrf9160_gps.gps_thread: Waiting for time window to operate
[00:00:03.973,541] <inf> cat_tracker: GPS_EVT_OPERATION_BLOCKED
[00:00:05.800,537] <dbg> cat_tracker.modem_rsrp_handler: Incoming RSRP status message, RSRP value is 38
[00:00:08.223,785] <dbg> cat_tracker.lte_evt_handler: RRC mode: Idle
[00:00:08.230,926] <dbg> nrf9160_gps.gps_thread: GPS has time window to operate
[00:00:08.238,800] <inf> cat_tracker: GPS_EVT_OPERATION_UNBLOCKED
[00:00:08.245,483] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:08.254,608] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 8
[00:00:09.227,478] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:09.236,602] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 9
[00:00:10.227,508] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:10.236,602] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 10
[00:00:11.227,539] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:11.236,663] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 11
[00:00:12.246,337] <dbg> cat_tracker.modem_rsrp_handler: Incoming RSRP status message, RSRP value is 39
[00:00:12.256,561] <dbg> cat_tracker.modem_rsrp_handler: Incoming RSRP status message, RSRP value is 39
[00:00:12.686,340] <dbg> cat_tracker.lte_evt_handler: RRC mode: Connected
[00:00:13.176,361] <dbg> nrf9160_gps.gps_thread: Waiting for time window to operate
[00:00:13.184,600] <inf> cat_tracker: GPS_EVT_OPERATION_BLOCKED
[00:00:13.218,597] <inf> cat_tracker: CLOUD_EVT_DISCONNECTED
[00:00:13.725,036] <inf> cat_tracker: CLOUD_EVT_DISCONNECTED
[00:00:18.062,225] <dbg> cat_tracker.lte_evt_handler: RRC mode: Idle
[00:00:18.069,396] <dbg> nrf9160_gps.gps_thread: GPS has time window to operate
[00:00:18.077,270] <inf> cat_tracker: GPS_EVT_OPERATION_UNBLOCKED
[00:00:18.083,953] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:18.093,078] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 18
[00:00:19.064,636] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:19.073,760] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 19
[00:00:20.064,514] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:20.073,638] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 20
[00:00:21.064,666] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:21.073,791] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 21
[00:00:22.064,758] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:22.073,883] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 22
[00:00:23.064,849] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:23.073,944] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 23
[00:00:24.064,636] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:24.073,760] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 24
[00:00:25.064,788] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:25.073,913] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 25
[00:00:26.064,483] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:26.073,577] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 26
[00:00:27.064,971] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:27.074,096] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 27
[00:00:28.064,727] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:28.073,852] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 28
[00:00:29.064,910] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:29.074,005] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 29
[00:00:30.064,575] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:30.073,699] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 30
[00:00:30.213,043] <dbg> watchdog.primary_feed_worker: Feeding watchdog
[00:00:31.064,605] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:31.073,730] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 31
[00:00:32.065,063] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:32.074,188] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 32
[00:00:33.064,666] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:33.073,791] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 33
[00:00:34.065,032] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:34.074,127] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 34
[00:00:35.064,727] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:35.073,852] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 35
[00:00:36.065,185] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:36.074,310] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 36
[00:00:37.065,155] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:37.074,279] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 37
[00:00:38.065,246] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:38.074,371] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 38
[00:00:39.065,032] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:39.074,157] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 39
[00:00:40.065,307] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:40.074,401] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 40
[00:00:41.065,277] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:41.074,401] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 41
[00:00:42.065,368] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:42.074,493] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 42
[00:00:43.065,368] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:43.074,493] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 43
[00:00:44.064,971] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:44.074,096] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 44
[00:00:45.065,338] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:45.074,432] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 45
[00:00:46.065,246] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:46.074,340] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 46
[00:00:47.065,368] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:47.074,493] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 47
[00:00:48.065,399] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:48.074,523] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 48
[00:00:49.065,093] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:49.074,188] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 49
[00:00:50.065,307] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:50.074,432] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 50
[00:00:51.065,521] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:51.074,615] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 51
[00:00:52.065,368] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:52.074,493] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 52
[00:00:53.065,551] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:53.074,645] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 53
[00:00:54.065,551] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:54.074,645] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 54
[00:00:55.065,704] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:55.074,798] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 55
[00:00:56.065,704] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:56.074,829] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 56
[00:00:57.065,643] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:57.074,859] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 57
[00:00:58.065,429] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:58.074,554] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 58
[00:00:59.065,368] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:00:59.074,493] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 59
[00:01:00.065,765] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:01:00.074,981] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 60
[00:01:00.220,306] <dbg> watchdog.primary_feed_worker: Feeding watchdog
[00:01:01.065,429] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:01:01.074,554] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 61
[00:01:02.065,460] <dbg> nrf9160_gps.print_satellite_stats: Tracking: 0 Using: 0 Unhealthy: 0
[00:01:02.074,584] <dbg> nrf9160_gps.print_satellite_stats: Seconds since last fix 62
[00:01:04.083,251] <inf> cat_tracker: GPS_EVT_SEARCH_TIMEOUT
[00:01:04.089,599] <dbg> nrf9160_gps.stop_gps: Stopping GPS
[00:01:04.104,553] <inf> cat_tracker: GPS_EVT_SEARCH_STOPPED
[00:01:04.110,961] <inf> cat_tracker: Not connected to cloud!
[00:01:04.117,431] <inf> cat_tracker: Going to sleep for: 60 seconds
[00:01:04.234,741] <inf> cat_tracker: Board is running expected modem firmware version: mfw_nrf9160_1.2.1
[00:01:04.244,934] <dbg> cat_tracker.modem_buffer_populate: Entry: 1 of 19 in modem buffer filled
[00:01:04.254,394] <dbg> cat_tracker.battery_buffer_populate: Entry: 1 of 19 in battery buffer filled
[00:01:04.265,655] <dbg> cat_tracker.sensors_buffer_populate: Entry: 1 of 19 in sensor buffer filled

```

**Comment by simensrostad** [comment]
This will be handled by the reconnection routine in the new firmware architecture.

**Comment by coderbyheart** [comment]
This blocks #3 because the device will never connect to the cloud, since all tests use a fresh set of credentials

**Comment by coderbyheart** [comment]
Fixed in a90d0057d6046a1d0683f8f935f22ab1cf9820e0
